### PR TITLE
Detect pass alert/v2

### PR DESCRIPTION
**Pull request by victorjulien**
This implements an `alert` keyword, as the opposite of the `noalert` keyword. It allows rules that `pass`-then-`alert`.

https://redmine.openinfosecfoundation.org/issues/5466

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1579

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ASAN_TLPR1_suri.
  

Pipeline 17490

File: src/detect-engine-alert.c [hunk 1]
```diff
@@ -407,22 +404,30 @@ void PacketAlertFinalize(DetectEngineCtx *de_ctx, DetectEngineThreadCtx *det_ctx
...
             p->alerts.suppressed++;
         } else if (p->alerts.cnt < packet_alert_max) {
             p->alerts.alerts[p->alerts.cnt] = *pa;
             SCLogDebug("Appending sid %" PRIu32 " alert to Packet::alerts at pos %u", s->id, i);
 
-            /* pass "alert" found, we're done */
-            if (pa->action & ACTION_PASS) {
+            /* pass-alert found, we're done. Alert is not logged. */
+            if ((pa->action & (ACTION_PASS | ACTION_ALERT)) == ACTION_PASS) {
+                SCLogDebug("sid:%u: is a pass rule, so break out of out loop", s->id);
```
Comment (jufajardini) [inline src/detect-engine-alert.c:418]: typo? -> "out of out loop"

File: src/detect-noalert.c [hunk 2]
```diff
@@ -20,33 +20,43 @@
...
-    if (nullstr != NULL) {
-        SCLogError("nocase has no value");
-        return -1;
-    }
-
-    s->flags |= SIG_FLAG_NOALERT;
-    return 0;
+    sigmatch_table[DETECT_ALERT].name = "alert";
+    sigmatch_table[DETECT_ALERT].desc = "alert will be generated by the rule";
+    sigmatch_table[DETECT_ALERT].url = "/rules/flow-keywords.html";
```
Comment (jufajardini) [inline src/detect-noalert.c:59]: Looks like we need to document this new behavior, and possibly also properly document `noalert`. We do explain it, but only in passing, when talking about other keywords.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW1_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .uptime | 137 | 146 | 106.57% | 

Pipeline 17491

**Review by jufajardini — COMMENTED** [review]
Looking good so far, just two inline minor comments.

File: src/detect-noalert.c (hunk 2 above)
Comment (jufajardini) [reply src/detect-noalert.c:59]: Added ticket to track this: https://redmine.openinfosecfoundation.org/issues/6685

**Comment by catenacyber** [comment]
Any update on this ?

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on SURI_TLPW1_suri_time.

ERROR: QA failed on SURI_TLPW1_stats_chk.
  

Pipeline 17491

**Comment by victorjulien** [comment]
replaced by #10157

**Pull request closed without merging**